packages/core/test/controls/numericInputTests.tsx

```diff
@@ -381,6 +392,18 @@ describe("<NumericInput>", () => {
                 const newValue = component.state().value;
                 expect(newValue).to.equal(MIN_VALUE.toString());
             });
+
+            it("fires onValueChange with clamped value if props.min was updated to exceed current value ", () => {
+                const onValueChangeSpy = sinon.spy();
+                const component = mount(<NumericInput value={-10} onValueChange={onValueChangeSpy} />);
+
+                component.setProps({ min: 0 });
+
+                const newValue = component.state().value;
+                expect(newValue).to.equal("0");
+                expect(onValueChangeSpy.calledOnce).to.be.true;
+                expect(onValueChangeSpy.firstCall.args).to.deep.equal([0, "0"]);
+            });
```

Comment:
+ it shouldn't change if `props.min` updated to be less than current value
and the same below